packages/kbn-custom-integrations/src/hooks/use_consumer_custom_integrations.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useMemo } from 'react';
+import {
+  CreateDispatchableEvents,
+  useCreateDispatchableEvents,
+} from './create/use_create_dispatchable_events';
+import { useCustomIntegrations } from './use_custom_integrations';
+
+export const useConsumerCustomIntegrations = () => {
+  const { customIntegrationsState } = useCustomIntegrations();
+  const createDispatchableEvents = useCreateDispatchableEvents({
+    machineRef: customIntegrationsState.children.createCustomIntegration,
+  });
+
+  const dispatchableEvents = useMemo(() => {
+    return {
+      ...createDispatchableEvents,
+    };
+  }, [createDispatchableEvents]);
```

Comment:
**nit:** The return value from `useCreateDispatchableEvents` is already memorized with a useMemo, this new memorized object could be removed and `dispatchableEvents` can be named after the hook return value directly:
```ts
const dispatchableEvents = useCreateDispatchableEvents({
  machineRef: customIntegrationsState.children.createCustomIntegration,
});

return {
  mode: customIntegrationsState.context.mode,
  dispatchableEvents,
};
```

Reply:
Oh, good spot. I imagine this got lost amongst the many mini-refactors I ended up doing 😂

Reply:
Oh actually, I think this was just me planning ahead for multiple modes. But I'll change it for now for the very small performance increase.